kedro/contrib/io/parquet/parquet_s3.py

```diff
@@ -0,0 +1,160 @@
+            >>>
+            >>> data_set = ParquetS3DataSet(
+            >>>                         filepath="temp3.parquet",
+            >>>                         bucket_name="test_bucket",
+            >>>                         credentials={
+            >>>                             'aws_access_key_id': 'YOUR_KEY',
+            >>>                             'aws_access_secredt_key': 'YOUR SECRET'},
+            >>>                         save_args={"compression": "GZIP"})
+            >>> data_set.save(data)
+            >>> reloaded = data_set.load()
+            >>>
+            >>> assert data.equals(reloaded)
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        bucket_name: str,
+        credentials: Optional[Dict[str, Any]] = None,
+        load_args: Optional[Dict[str, Any]] = None,
+        save_args: Optional[Dict[str, Any]] = None,
+        version: Version = None,
+    ) -> None:
+        """Creates a new instance of ``ParquetS3DataSet`` pointing to a concrete
+        parquet file on S3.
+
+        Args:
+            filepath: Path to a parquet file
+                parquet collection or the directory of a multipart parquet.
+            bucket_name: S3 bucket name.
+            credentials: Credentials to access the S3 bucket, such as
+                ``aws_access_key_id``, ``aws_secret_access_key``.
+            load_args: Additional loading options `pyarrow`:
+                https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html
+                or `fastparquet`:
+                https://fastparquet.readthedocs.io/en/latest/api.html#fastparquet.ParquetFile.to_pandas
+            save_args: Additional saving options for `pyarrow`:
+                https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.from_pandas
+                or `fastparquet`:
+                https://fastparquet.readthedocs.io/en/latest/api.html#fastparquet.write
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+        """
+
+        _credentials = deepcopy(credentials) or {}
+        _s3 = S3FileSystem(client_kwargs=_credentials)
+        super().__init__(
+            PurePosixPath("{}/{}".format(bucket_name, filepath)),
+            version,
+            exists_function=_s3.exists,
+            glob_function=partial(_s3.glob, refresh=True),
+        )
+
+        default_load_args = {}  # type: Dict[str, Any]
+        default_save_args = {}  # type: Dict[str, Any]
+
+        self._load_args = (
+            {**default_load_args, **load_args}
+            if load_args is not None
+            else default_load_args
+        )
+        self._save_args = (
```

Comment:
`self._save_args` did not seem to be used anywhere.

Reply:
@921kiyo I just updated `save()` to use `self._save_args`.